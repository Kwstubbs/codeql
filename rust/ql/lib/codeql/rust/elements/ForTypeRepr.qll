// generated by codegen, do not edit
/**
 * This module provides the public class `ForTypeRepr`.
 */

private import internal.ForTypeReprImpl
import codeql.rust.elements.ForBinder
import codeql.rust.elements.TypeRepr

/**
 * A type with a higher-ranked `for` modifier. This is currently not valid Rust syntax (`for<...>` can
 * only be applied to traits to form a `TypeBound`).
 *
 * For example:
 * ```rust
 * fn foo(value: for<'a> usize) {}  // DOESN'T COMPILE
 * //            ^^^^^^^^^^^^^
 * ```
 */
final class ForTypeRepr = Impl::ForTypeRepr;
