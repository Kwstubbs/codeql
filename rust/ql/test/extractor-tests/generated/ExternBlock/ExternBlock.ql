// generated by codegen, do not edit
import codeql.rust.elements
import TestUtils

query predicate instances(
  ExternBlock x, string hasExtendedCanonicalPath__label, string hasExtendedCanonicalPath,
  string hasCrateOrigin__label, string hasCrateOrigin, string hasAttributeMacroExpansion__label,
  string hasAttributeMacroExpansion, string hasAbi__label, string hasAbi,
  string getNumberOfAttrs__label, int getNumberOfAttrs, string hasExternItemList__label,
  string hasExternItemList, string isUnsafe__label, string isUnsafe
) {
  toBeTested(x) and
  not x.isUnknown() and
  hasExtendedCanonicalPath__label = "hasExtendedCanonicalPath:" and
  (
    if x.hasExtendedCanonicalPath()
    then hasExtendedCanonicalPath = "yes"
    else hasExtendedCanonicalPath = "no"
  ) and
  hasCrateOrigin__label = "hasCrateOrigin:" and
  (if x.hasCrateOrigin() then hasCrateOrigin = "yes" else hasCrateOrigin = "no") and
  hasAttributeMacroExpansion__label = "hasAttributeMacroExpansion:" and
  (
    if x.hasAttributeMacroExpansion()
    then hasAttributeMacroExpansion = "yes"
    else hasAttributeMacroExpansion = "no"
  ) and
  hasAbi__label = "hasAbi:" and
  (if x.hasAbi() then hasAbi = "yes" else hasAbi = "no") and
  getNumberOfAttrs__label = "getNumberOfAttrs:" and
  getNumberOfAttrs = x.getNumberOfAttrs() and
  hasExternItemList__label = "hasExternItemList:" and
  (if x.hasExternItemList() then hasExternItemList = "yes" else hasExternItemList = "no") and
  isUnsafe__label = "isUnsafe:" and
  if x.isUnsafe() then isUnsafe = "yes" else isUnsafe = "no"
}

query predicate getExtendedCanonicalPath(ExternBlock x, string getExtendedCanonicalPath) {
  toBeTested(x) and not x.isUnknown() and getExtendedCanonicalPath = x.getExtendedCanonicalPath()
}

query predicate getCrateOrigin(ExternBlock x, string getCrateOrigin) {
  toBeTested(x) and not x.isUnknown() and getCrateOrigin = x.getCrateOrigin()
}

query predicate getAttributeMacroExpansion(ExternBlock x, MacroItems getAttributeMacroExpansion) {
  toBeTested(x) and
  not x.isUnknown() and
  getAttributeMacroExpansion = x.getAttributeMacroExpansion()
}

query predicate getAbi(ExternBlock x, Abi getAbi) {
  toBeTested(x) and not x.isUnknown() and getAbi = x.getAbi()
}

query predicate getAttr(ExternBlock x, int index, Attr getAttr) {
  toBeTested(x) and not x.isUnknown() and getAttr = x.getAttr(index)
}

query predicate getExternItemList(ExternBlock x, ExternItemList getExternItemList) {
  toBeTested(x) and not x.isUnknown() and getExternItemList = x.getExternItemList()
}
