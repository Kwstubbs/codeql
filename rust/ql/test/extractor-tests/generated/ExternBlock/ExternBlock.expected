instances
| gen_extern_block.rs:7:5:9:5 | ExternBlock | hasExtendedCanonicalPath: | no | hasCrateOrigin: | no | hasAttributeMacroExpansion: | no | hasAbi: | yes | getNumberOfAttrs: | 0 | hasExternItemList: | yes | isUnsafe: | no |
getExtendedCanonicalPath
getCrateOrigin
getAttributeMacroExpansion
getAbi
| gen_extern_block.rs:7:5:9:5 | ExternBlock | gen_extern_block.rs:7:5:7:14 | Abi |
getAttr
getExternItemList
| gen_extern_block.rs:7:5:9:5 | ExternBlock | gen_extern_block.rs:7:16:9:5 | ExternItemList |
